### PR TITLE
Use 2021-10-01-preview for stream analytics outputs api

### PR DESCRIPTION
**Pull request by marcgs**
Update stream analytics output to version `2021-10-01-preview`. This version is required in follow-up PRs in order to support optional shared access policy:
- #19712 
- #19708 

Acceptance tests for stream analytics outputs were run and passed:
```
TF_ACC=1 go test -v ./internal/services/streamanalytics -run=TestAccStreamAnalyticsOutput -timeout 40m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccStreamAnalyticsOutputBlob_avro
=== PAUSE TestAccStreamAnalyticsOutputBlob_avro
=== RUN   TestAccStreamAnalyticsOutputBlob_csv
=== PAUSE TestAccStreamAnalyticsOutputBlob_csv
=== RUN   TestAccStreamAnalyticsOutputBlob_json
=== PAUSE TestAccStreamAnalyticsOutputBlob_json
=== RUN   TestAccStreamAnalyticsOutputBlob_parquet
=== PAUSE TestAccStreamAnalyticsOutputBlob_parquet
=== RUN   TestAccStreamAnalyticsOutputBlob_update
=== PAUSE TestAccStreamAnalyticsOutputBlob_update
=== RUN   TestAccStreamAnalyticsOutputBlob_requiresImport
=== PAUSE TestAccStreamAnalyticsOutputBlob_requiresImport
=== RUN   TestAccStreamAnalyticsOutputBlob_authenticationMode
=== PAUSE TestAccStreamAnalyticsOutputBlob_authenticationMode
=== RUN   TestAccStreamAnalyticsOutputCosmosDB_basic
=== PAUSE TestAccStreamAnalyticsOutputCosmosDB_basic
=== RUN   TestAccStreamAnalyticsOutputCosmosDB_update
=== PAUSE TestAccStreamAnalyticsOutputCosmosDB_update
=== RUN   TestAccStreamAnalyticsOutputCosmosDB_complete
=== PAUSE TestAccStreamAnalyticsOutputCosmosDB_complete
=== RUN   TestAccStreamAnalyticsOutputCosmosDB_requiresImport
=== PAUSE TestAccStreamAnalyticsOutputCosmosDB_requiresImport
=== RUN   TestAccStreamAnalyticsOutputEventHub_avro
=== PAUSE TestAccStreamAnalyticsOutputEventHub_avro
=== RUN   TestAccStreamAnalyticsOutputEventHub_csv
=== PAUSE TestAccStreamAnalyticsOutputEventHub_csv
=== RUN   TestAccStreamAnalyticsOutputEventHub_json
=== PAUSE TestAccStreamAnalyticsOutputEventHub_json
=== RUN   TestAccStreamAnalyticsOutputEventHub_jsonArrayFormat
=== PAUSE TestAccStreamAnalyticsOutputEventHub_jsonArrayFormat
=== RUN   TestAccStreamAnalyticsOutputEventHub_propertyColumns
=== PAUSE TestAccStreamAnalyticsOutputEventHub_propertyColumns
=== RUN   TestAccStreamAnalyticsOutputEventHub_partitionKey
=== PAUSE TestAccStreamAnalyticsOutputEventHub_partitionKey
=== RUN   TestAccStreamAnalyticsOutputEventHub_update
=== PAUSE TestAccStreamAnalyticsOutputEventHub_update
=== RUN   TestAccStreamAnalyticsOutputEventHub_requiresImport
=== PAUSE TestAccStreamAnalyticsOutputEventHub_requiresImport
=== RUN   TestAccStreamAnalyticsOutputEventHub_authenticationMode
=== PAUSE TestAccStreamAnalyticsOutputEventHub_authenticationMode
=== RUN   TestAccStreamAnalyticsOutputFunction_basic
=== PAUSE TestAccStreamAnalyticsOutputFunction_basic
=== RUN   TestAccStreamAnalyticsOutputFunction_complete
=== PAUSE TestAccStreamAnalyticsOutputFunction_complete
=== RUN   TestAccStreamAnalyticsOutputFunction_update
=== PAUSE TestAccStreamAnalyticsOutputFunction_update
=== RUN   TestAccStreamAnalyticsOutputFunction_requiresImport
=== PAUSE TestAccStreamAnalyticsOutputFunction_requiresImport
=== RUN   TestAccStreamAnalyticsOutputSql_basic
=== PAUSE TestAccStreamAnalyticsOutputSql_basic
=== RUN   TestAccStreamAnalyticsOutputSql_update
=== PAUSE TestAccStreamAnalyticsOutputSql_update
=== RUN   TestAccStreamAnalyticsOutputSql_authenticationModeMsi
=== PAUSE TestAccStreamAnalyticsOutputSql_authenticationModeMsi
=== RUN   TestAccStreamAnalyticsOutputSql_requiresImport
=== PAUSE TestAccStreamAnalyticsOutputSql_requiresImport
=== RUN   TestAccStreamAnalyticsOutputSql_maxBatchCountAndMaxWriterCount
=== PAUSE TestAccStreamAnalyticsOutputSql_maxBatchCountAndMaxWriterCount
=== RUN   TestAccStreamAnalyticsOutputPowerBI_basic
=== PAUSE TestAccStreamAnalyticsOutputPowerBI_basic
=== RUN   TestAccStreamAnalyticsOutputPowerBI_update
=== PAUSE TestAccStreamAnalyticsOutputPowerBI_update
=== RUN   TestAccStreamAnalyticsOutputPowerBI_complete
=== PAUSE TestAccStreamAnalyticsOutputPowerBI_complete
=== RUN   TestAccStreamAnalyticsOutputServiceBusQueue_avro
=== PAUSE TestAccStreamAnalyticsOutputServiceBusQueue_avro
=== RUN   TestAccStreamAnalyticsOutputServiceBusQueue_csv
=== PAUSE TestAccStreamAnalyticsOutputServiceBusQueue_csv
=== RUN   TestAccStreamAnalyticsOutputServiceBusQueue_json
=== PAUSE TestAccStreamAnalyticsOutputServiceBusQueue_json
=== RUN   TestAccStreamAnalyticsOutputServiceBusQueue_authenticationMode
=== PAUSE TestAccStreamAnalyticsOutputServiceBusQueue_authenticationMode
=== RUN   TestAccStreamAnalyticsOutputServiceBusQueue_update
=== PAUSE TestAccStreamAnalyticsOutputServiceBusQueue_update
=== RUN   TestAccStreamAnalyticsOutputServiceBusQueue_requiresImport
=== PAUSE TestAccStreamAnalyticsOutputServiceBusQueue_requiresImport
=== RUN   TestAccStreamAnalyticsOutputServiceBusQueue_propertyColumns
=== PAUSE TestAccStreamAnalyticsOutputServiceBusQueue_propertyColumns
=== RUN   TestAccStreamAnalyticsOutputServiceBusQueue_systemPropertyColumns
=== PAUSE TestAccStreamAnalyticsOutputServiceBusQueue_systemPropertyColumns
=== RUN   TestAccStreamAnalyticsOutputServiceBusTopic_avro
=== PAUSE TestAccStreamAnalyticsOutputServiceBusTopic_avro
=== RUN   TestAccStreamAnalyticsOutputServiceBusTopic_csv
=== PAUSE TestAccStreamAnalyticsOutputServiceBusTopic_csv
=== RUN   TestAccStreamAnalyticsOutputServiceBusTopic_json
=== PAUSE TestAccStreamAnalyticsOutputServiceBusTopic_json
=== RUN   TestAccStreamAnalyticsOutputServiceBusTopic_propertyColumns
=== PAUSE TestAccStreamAnalyticsOutputServiceBusTopic_propertyColumns
=== RUN   TestAccStreamAnalyticsOutputServiceBusTopic_update
=== PAUSE TestAccStreamAnalyticsOutputServiceBusTopic_update
=== RUN   TestAccStreamAnalyticsOutputServiceBusTopic_requiresImport
=== PAUSE TestAccStreamAnalyticsOutputServiceBusTopic_requiresImport
=== RUN   TestAccStreamAnalyticsOutputServiceBusTopic_authenticationMode
=== PAUSE TestAccStreamAnalyticsOutputServiceBusTopic_authenticationMode
=== RUN   TestAccStreamAnalyticsOutputServiceBusTopic_systemPropertyColumns
=== PAUSE TestAccStreamAnalyticsOutputServiceBusTopic_systemPropertyColumns
=== RUN   TestAccStreamAnalyticsOutputSynapse_basic
=== PAUSE TestAccStreamAnalyticsOutputSynapse_basic
=== RUN   TestAccStreamAnalyticsOutputSynapse_sqlPool
=== PAUSE TestAccStreamAnalyticsOutputSynapse_sqlPool
=== RUN   TestAccStreamAnalyticsOutputSynapse_update
=== PAUSE TestAccStreamAnalyticsOutputSynapse_update
=== RUN   TestAccStreamAnalyticsOutputSynapse_requiresImport
=== PAUSE TestAccStreamAnalyticsOutputSynapse_requiresImport
=== RUN   TestAccStreamAnalyticsOutputTable_basic
=== PAUSE TestAccStreamAnalyticsOutputTable_basic
=== RUN   TestAccStreamAnalyticsOutputTable_update
=== PAUSE TestAccStreamAnalyticsOutputTable_update
=== RUN   TestAccStreamAnalyticsOutputTable_requiresImport
=== PAUSE TestAccStreamAnalyticsOutputTable_requiresImport
=== RUN   TestAccStreamAnalyticsOutputTable_columnsToRemove
=== PAUSE TestAccStreamAnalyticsOutputTable_columnsToRemove
=== CONT  TestAccStreamAnalyticsOutputBlob_avro
=== CONT  TestAccStreamAnalyticsOutputSql_maxBatchCountAndMaxWriterCount
=== CONT  TestAccStreamAnalyticsOutputEventHub_jsonArrayFormat
=== CONT  TestAccStreamAnalyticsOutputSql_requiresImport
=== CONT  TestAccStreamAnalyticsOutputSql_authenticationModeMsi
=== CONT  TestAccStreamAnalyticsOutputSql_update
=== CONT  TestAccStreamAnalyticsOutputSql_basic
=== CONT  TestAccStreamAnalyticsOutputFunction_requiresImport
=== CONT  TestAccStreamAnalyticsOutputFunction_update
=== CONT  TestAccStreamAnalyticsOutputFunction_complete
=== CONT  TestAccStreamAnalyticsOutputFunction_basic
=== CONT  TestAccStreamAnalyticsOutputEventHub_authenticationMode
=== CONT  TestAccStreamAnalyticsOutputEventHub_requiresImport
=== CONT  TestAccStreamAnalyticsOutputEventHub_update
=== CONT  TestAccStreamAnalyticsOutputServiceBusTopic_json
=== CONT  TestAccStreamAnalyticsOutputEventHub_partitionKey
=== CONT  TestAccStreamAnalyticsOutputTable_columnsToRemove
--- PASS: TestAccStreamAnalyticsOutputBlob_avro (141.97s)
--- PASS: TestAccStreamAnalyticsOutputFunction_requiresImport (211.00s)
=== CONT  TestAccStreamAnalyticsOutputTable_requiresImport
--- PASS: TestAccStreamAnalyticsOutputFunction_basic (269.01s)
=== CONT  TestAccStreamAnalyticsOutputTable_update
--- PASS: TestAccStreamAnalyticsOutputFunction_complete (269.97s)
=== CONT  TestAccStreamAnalyticsOutputTable_basic
--- PASS: TestAccStreamAnalyticsOutputEventHub_partitionKey (289.16s)
=== CONT  TestAccStreamAnalyticsOutputSynapse_requiresImport
--- PASS: TestAccStreamAnalyticsOutputServiceBusTopic_json (289.34s)
=== CONT  TestAccStreamAnalyticsOutputSynapse_update
--- PASS: TestAccStreamAnalyticsOutputEventHub_authenticationMode (292.71s)
=== CONT  TestAccStreamAnalyticsOutputEventHub_propertyColumns
--- PASS: TestAccStreamAnalyticsOutputSql_authenticationModeMsi (296.70s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusTopic_authenticationMode
--- PASS: TestAccStreamAnalyticsOutputEventHub_requiresImport (310.50s)
=== CONT  TestAccStreamAnalyticsOutputSynapse_basic
--- PASS: TestAccStreamAnalyticsOutputFunction_update (311.06s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusTopic_systemPropertyColumns
--- PASS: TestAccStreamAnalyticsOutputSql_requiresImport (312.44s)
=== CONT  TestAccStreamAnalyticsOutputSynapse_sqlPool
--- PASS: TestAccStreamAnalyticsOutputEventHub_jsonArrayFormat (316.90s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusTopic_requiresImport
--- PASS: TestAccStreamAnalyticsOutputSql_basic (327.05s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusTopic_update
--- PASS: TestAccStreamAnalyticsOutputSql_update (357.95s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusTopic_propertyColumns
--- PASS: TestAccStreamAnalyticsOutputTable_requiresImport (154.89s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusQueue_authenticationMode
--- PASS: TestAccStreamAnalyticsOutputTable_basic (137.29s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusQueue_propertyColumns
--- PASS: TestAccStreamAnalyticsOutputEventHub_update (427.75s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusQueue_requiresImport
--- PASS: TestAccStreamAnalyticsOutputTable_update (216.65s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusQueue_update
--- PASS: TestAccStreamAnalyticsOutputTable_columnsToRemove (349.93s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusQueue_systemPropertyColumns
--- PASS: TestAccStreamAnalyticsOutputSql_maxBatchCountAndMaxWriterCount (519.66s)
=== CONT  TestAccStreamAnalyticsOutputCosmosDB_basic
--- PASS: TestAccStreamAnalyticsOutputEventHub_propertyColumns (282.35s)
=== CONT  TestAccStreamAnalyticsOutputEventHub_json
--- PASS: TestAccStreamAnalyticsOutputServiceBusTopic_authenticationMode (290.35s)
=== CONT  TestAccStreamAnalyticsOutputEventHub_csv
--- PASS: TestAccStreamAnalyticsOutputServiceBusTopic_requiresImport (315.39s)
=== CONT  TestAccStreamAnalyticsOutputEventHub_avro
--- PASS: TestAccStreamAnalyticsOutputServiceBusTopic_propertyColumns (288.82s)
=== CONT  TestAccStreamAnalyticsOutputCosmosDB_requiresImport
--- PASS: TestAccStreamAnalyticsOutputServiceBusTopic_update (423.32s)
=== CONT  TestAccStreamAnalyticsOutputBlob_authenticationMode
--- PASS: TestAccStreamAnalyticsOutputSynapse_basic (486.04s)
=== CONT  TestAccStreamAnalyticsOutputCosmosDB_complete
--- PASS: TestAccStreamAnalyticsOutputServiceBusQueue_authenticationMode (433.91s)
=== CONT  TestAccStreamAnalyticsOutputBlob_update
--- PASS: TestAccStreamAnalyticsOutputServiceBusTopic_systemPropertyColumns (519.70s)
=== CONT  TestAccStreamAnalyticsOutputCosmosDB_update
--- PASS: TestAccStreamAnalyticsOutputSynapse_update (554.00s)
=== CONT  TestAccStreamAnalyticsOutputBlob_requiresImport
--- PASS: TestAccStreamAnalyticsOutputSynapse_requiresImport (567.33s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusTopic_csv
--- PASS: TestAccStreamAnalyticsOutputEventHub_json (286.66s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusTopic_avro
=== CONT  TestAccStreamAnalyticsOutputPowerBI_update
--- PASS: TestAccStreamAnalyticsOutputServiceBusQueue_requiresImport (451.75s)
--- PASS: TestAccStreamAnalyticsOutputEventHub_csv (299.76s)
=== CONT  TestAccStreamAnalyticsOutputPowerBI_complete
--- PASS: TestAccStreamAnalyticsOutputEventHub_avro (288.11s)
=== CONT  TestAccStreamAnalyticsOutputPowerBI_basic
--- PASS: TestAccStreamAnalyticsOutputBlob_authenticationMode (210.26s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusQueue_csv
--- PASS: TestAccStreamAnalyticsOutputPowerBI_complete (94.03s)
=== CONT  TestAccStreamAnalyticsOutputBlob_json
--- PASS: TestAccStreamAnalyticsOutputBlob_requiresImport (153.71s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusQueue_json
--- PASS: TestAccStreamAnalyticsOutputBlob_update (210.19s)
=== CONT  TestAccStreamAnalyticsOutputServiceBusQueue_avro
--- PASS: TestAccStreamAnalyticsOutputPowerBI_basic (103.16s)
=== CONT  TestAccStreamAnalyticsOutputBlob_csv
--- PASS: TestAccStreamAnalyticsOutputPowerBI_update (164.81s)
=== CONT  TestAccStreamAnalyticsOutputBlob_parquet
--- PASS: TestAccStreamAnalyticsOutputServiceBusQueue_propertyColumns (646.95s)
--- PASS: TestAccStreamAnalyticsOutputBlob_json (134.71s)
--- PASS: TestAccStreamAnalyticsOutputSynapse_sqlPool (813.51s)
--- PASS: TestAccStreamAnalyticsOutputServiceBusQueue_systemPropertyColumns (637.15s)
--- PASS: TestAccStreamAnalyticsOutputServiceBusTopic_csv (285.41s)
--- PASS: TestAccStreamAnalyticsOutputServiceBusTopic_avro (289.23s)
--- PASS: TestAccStreamAnalyticsOutputBlob_csv (129.25s)
--- PASS: TestAccStreamAnalyticsOutputBlob_parquet (133.23s)
--- PASS: TestAccStreamAnalyticsOutputServiceBusQueue_update (704.49s)
--- PASS: TestAccStreamAnalyticsOutputServiceBusQueue_csv (420.72s)
--- PASS: TestAccStreamAnalyticsOutputCosmosDB_basic (874.50s)
--- PASS: TestAccStreamAnalyticsOutputServiceBusQueue_json (434.79s)
--- PASS: TestAccStreamAnalyticsOutputServiceBusQueue_avro (431.28s)
--- PASS: TestAccStreamAnalyticsOutputCosmosDB_requiresImport (868.11s)
--- PASS: TestAccStreamAnalyticsOutputCosmosDB_complete (847.61s)
--- PASS: TestAccStreamAnalyticsOutputCosmosDB_update (970.50s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/streamanalytics	1803.100s
```

### CODE REVIEW

**Review by katbyte — APPROVED** [review]
Thanks @marcgs - LGTM once the merge conflicts are resolved! 🏞️

**Comment by tombuildsstuff** [comment]
@marcgs hope you don't mind, pushed a commit to fix the remaining build issues so that we can get this merged

**Review by tombuildsstuff — APPROVED** [review]

**Comment by marcgs** [comment]
> @marcgs hope you don't mind, pushed a commit to fix the remaining build issues so that we can get this merged

Thanks a lot for your support @tombuildsstuff . I will go ahead and merge the PR.

**Pull request merged**
